### ISSUE TITLE
Screen bars swapped until after a restart Qtile

### ISSUE THREAD

**Issue by TheClaus** [qtile/qtile#3148]
When I start up Qtile the screen bars are switched. If I restart Qtile using mod shift R then the bars will go be correct.

Using qtile 0.19.0-1 on Arch Linux.

It was working before I upgraded.

Here is my [.config](https://pastebin.com/x7sSwts4). I don't think I am doing anything fancy.

I have made sure that my monitors are setup properly per the arch wiki page for multihead.

I am currently using xrandr in the autostart.sh file.
`xrandr --output DVI-D-0 --off --output HDMI-0 --primary --mode 2560x1440 --pos 2560x0 --rotate normal --output DP-0 --mode 2560x1440 --pos 0x0 --rotate normal --output DP-1 --off &`

I have tried using nvidia-settings as well in my autostart.sh file.
`nvidia-settings --assign CurrentMetaMode="DPY-2: nvidia-auto-select +0+0 {ForceCompositionPipeline=On}, DPY-1: nvidia-auto-select +2560+0 {ForceCompositionPipeline=On}"`

I have also tried adding it in my xorg.conf.d/10-monitor.conf file.
```
Section "Monitor"
        Identifier      "HDMI-0"
        Option          "Primary" "true"
EndSection

Section "Monitor"
        Identifier      "DP-0"
        Option          "LeftOf" "HDMI-0"
EndSection
```
None of these seem to make a difference.

**Comment by elParaguayo** [comment]
I am not remotely an expert when it comes to multiple screens but I would have thought that running `xrandr` _after_ qtile has started would be too late for the configuration (although I appreciate that you say this was working before).

I would have thought you either want to run xrandr before starting qtile (e.g. in a `.xinit` file) or force the reload of the config after running xrandr in your startup hook.

**Comment by elParaguayo** [comment]
I'm pretty sure this is an issue with the `screen_change` hook (which is what is set when you use `reconfigure_screens = True` in your config).

This hook fires too early.

If I post a potential fix as a PR, can you test it for me?

**Comment by cr1ogen** [comment]
When i start qtile my bar look like this qtile 0.19.1.dev9+g4640369c

![1_grim](https://user-images.githubusercontent.com/9537788/147721835-e250c7e6-5283-4409-b5bb-f31553204e1e.png)

**Comment by cr1ogen** [comment]
If I restart Qtile using mod shift R the bar is seen in the correct position 
![2_grim](https://user-images.githubusercontent.com/9537788/147721968-5b61bdd2-1979-4905-9bf7-dfbb4df58edb.png)

**Comment by elParaguayo** [comment]
Are you able to test the potential fix in #3151?

**Comment by cr1ogen** [comment]
> Are you able to test the potential fix in #3151?
I'm sorry but I don't understand what to do to test it, but if it helps, that problem happens when in my autostart.sh file I have the following line active (I use wayland)

`wlr-randr --output DP-2 --scale 2 &

`

**Comment by elParaguayo** [comment]
Ah. If you're on Wayland then my fix has nothing to do with your issue.

Hmm. Maybe there's something else wrong. Will do some more digging.

**Comment by PucklaMotzer09** [comment]
I am experiencing the same issue. Everything was fine on `qtile 0.18.1`, but the bars are switched in `qtile 0.19.0`.
Main Monitor (right):

![2021-12-28_13-43_1](https://user-images.githubusercontent.com/25499721/148215190-619c51bc-fec2-45cb-aa99-3ff4254d578c.png)

Side Monitor (left):

![2021-12-28_13-43](https://user-images.githubusercontent.com/25499721/148215238-67011ded-b801-4681-b503-cac31ace4a11.png)

For me the issue is even weirder, because the screen resolution differs.
If you point in the right direction for testing your PR, I would gladly assist you, but I currently don't know how I safely switch my system qtile for a manually built one.

**Comment by m-col** [comment]
@cr1ogen I think your bug is unrelated to this one. Please could you open a separate issue for it, with your monitor setup?

**Comment by cr1ogen** [comment]
> 

Yes sure, no problem

**Comment by elParaguayo** [comment]
Think I've narrowed down the issue. Might take a but more time to fix this properly rather than a quick patch.
